### ISSUE TITLE
Network change back to WiFi from mobile does not trigger network listener/promise

### ISSUE THREAD

**Issue by lazmeister** [ionic-team/capacitor-plugins#1414]
### Plugin(s)
@capacitor/network

### Capacitor Version
4.1.0

### Platform(s)
Angular 14.2.8
Ionic/Angular 6.36
Web/PWA

### Current Behavior
When using Network.addListener, the promise is triggered when device is taken off Wi-Fi, though when you enable Wi-Fi afterwards, the listener does not get triggered.

### Expected Behavior
Network.addListener should be triggered when Wi-Fi is enabled or disabled.

### Code Reproduction

```
// Listen to network changes
      Network.addListener('networkStatusChange', status => {
        console.log(status.connected);
        if (status.connectionType === 'wifi') {
          networkConnectionWifi = true;
        } else {
          networkConnectionWifi = false;
        }
        // Check connection status
        currentNetworkStatus = status.connected;
      });
```

![image](https://user-images.githubusercontent.com/3281599/217584898-d9712cf9-3a85-45aa-b943-9a5d122888ed.png)

Here is the picture of the testing from the repo showing going from wifi to mobile connection, the network listener is triggered, but when going from mobile back to wifi the network listener does not trigger any event
image

public repo for testing: https://github.com/lazmeister/CapacitorWifiIssue

**Comment by lazmeister** [comment]
can anyone confirm they see the same results?

**Comment by mattsputnikdigital** [comment]
@lazmeister I have has some issues with this not updating when you would expect.

**Comment by gbrits** [comment]
I'm finding this plugin only triggers the listener once you turn the wifi off & back on again... which means you can't use the networkStatusChange event to indicate connectivity loss, only reconnection

**Comment by mattsputnikdigital** [comment]
@gbrits yeah I used the native browser window.navigator.onLine / events as a backup. I just have them all fire and then check window.navigator.onLine

**Comment by gbrits** [comment]
@mattsputnikdigital what about on iOS / Android though? :\

**Comment by mattsputnikdigital** [comment]
@gbrits Capacitor is running in a web view so it works on iOS / Android

**Comment by lazmeister** [comment]
@gbrits I remember reading on a Stackoverflow post where they were saying the only way to do this iOS/Android in the browser would be to use your own server with ping/transfers to find the speed. iOS/Safari has limited the network information for "privacy" reasons and that discussion is somewhere in webkit forums that has lasted for years now where requests have been put in but no solutions so far.

**Comment by eric-horodyski** [comment]
This thread has been inactive for some time, and we’re cleaning up older items to keep the backlog focused and manageable.

Details on the effort:
[https://ionic.io/blog/keeping-capacitors-backlog-healthy](https://ionic.io/blog/keeping-capacitors-backlog-healthy)

If this is still relevant today, feel free to open a new issue with current information and we’ll review it.
